currency/code.go

```diff
@@ -102,167 +100,55 @@ func (b *BaseCodes) GetCurrencies() Currencies {
 	return currencies
 }
 
-// UpdateCryptocurrency updates or registers a cryptocurrency
-func (b *BaseCodes) UpdateCryptocurrency(fullName, symbol string, id int) error {
+// UpdateCurrency updates or registers a currency/contract
+func (b *BaseCodes) UpdateCurrency(fullName, symbol, blockchain string, id int, r Role) error {
```

Comment:
UpdateCurrency() with the new changes only ever returns nil could be worth dropping the return value?